pyomo/core/tests/unit/test_sos_v2.py

```diff
@@ -0,0 +1,613 @@
+            (1, -7.5000000000e-02, 1e-3, False, 4),
+            (2, 1.1, 1e-3, True, 4),
+            (2, 1.1, 1e-3, False, 4),
+        ]
+
+        for sos, exp_res, abs_tol, use_rule, case in test_vectors:
+            try:
+                model = problem_indexed_sosn(case=case, n=sos, use_rule=use_rule)
+            except NotImplementedError:
+                continue
+
+            problem = model.create_instance()
+
+            opt.solve(problem, tee=show_output)
+
+            assert len(problem.mysos) != 0
+
+            assert math.isclose(pyo.value(problem.OBJ), exp_res, abs_tol=abs_tol)
+
+        # *********************************************************************
+
+    @unittest.skipIf(not solver_available, "The solver is not available.")
+    def test_nonindexedsos_multivar(self, show_output: bool = False):
+        "Test non-indexed SOS made up of different Var components."
+
+        opt = pyo.SolverFactory(solver_name)
+
+        # *********************************************************************
+
+        test_vectors = [
+            # sos, expected result, absolute tolerance
+            (1, 0.125, 1e-3),
+            (2, -0.07500000000000001, 1e-3),
+        ]
+
+        for sos, exp_res, abs_tol in test_vectors:
+            try:
+                model = problem_nonindexed_sosn_different_vars(n=sos)
+            except NotImplementedError:
+                continue
+
+            problem = model.create_instance()
+
+            opt.solve(problem, tee=show_output)
+
+            assert len(problem.mysos) != 0
+
+            assert math.isclose(pyo.value(problem.OBJ), exp_res, abs_tol=abs_tol)
+
+        # *********************************************************************
+
+    @unittest.skipIf(not solver_available, "The solver is not available.")
+    def test_indexedsos_multivar(self, show_output: bool = False):
+        "Test indexed SOS made up of different Var components."
+
+        opt = pyo.SolverFactory(solver_name)
+
+        # *********************************************************************
+
+        test_vectors = [
+            # sos, expected result, absolute tolerance
+            (1, -7.5000000000e-02, 1e-3),
+            (2, 1.1, 1e-3),
+        ]
+
+        for sos, exp_res, abs_tol in test_vectors:
+            try:
+                model = problem_indexed_sosn_different_vars(n=sos)
+            except NotImplementedError:
+                continue
+
+            problem = model.create_instance()
+
+            opt.solve(problem, tee=show_output)
+
+            assert len(problem.mysos) != 0
+
+            assert math.isclose(pyo.value(problem.OBJ), exp_res, abs_tol=abs_tol)
+
+        # *********************************************************************
+
+
+# *****************************************************************************
+# *****************************************************************************
+
+
+def problem_nonindexed_sosn_different_vars(n: int = 1):
```

Comment:
I don't care either way, but these methods can live inside the TestSOS class above.

Reply:
Done.